pkgs/development/python-modules/python-result/default.nix

```diff
@@ -0,0 +1,23 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+}:
+
+buildPythonPackage rec {
+  pname = "result";
+  version = "0.7.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-vWSUNfv0V94y5elB1j14fCEZOxJtSMhIRY3doWg96YA=";
+  };
+
+  pythonImportsCheck = [ "result" ];
+
```

Comment:
The package has tests we can run in addition to the import check.

Reply:
Having a hard time making it run.
Adding this
```
  checkInputs = [
    pytestCheckHook
  ];
```
gives me 
```
Executing pytestCheckPhase
ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: unrecognized arguments: --flake8 --cov=result --cov=tests --cov-report=term --cov-report=xml
  inifile: /build/result-0.7.0/pyproject.toml
  rootdir: /build/result-0.7.0
```

Reply:
Those lines https://github.com/rustedpy/result/blob/master/pyproject.toml#L37-L42 need to be dropped.

Reply:
Thank you! I'm trying this now:
```
  postPatch = ''
    substituteInPlace pyproject.toml \
      --replace '"--flake8",' "" \
      --replace '"--tb=short",' "" \
      --replace '"--cov=result",' "" \
      --replace '"--cov=tests",' "" \
      --replace '"--cov-report=term",' "" \
      --replace '"--cov-report=xml",' ""
  '';
```
It works, but it's quite verbose.

There is a problem here though. Github has `tests` folder, and its 0.7 release has tests, but Pypi package doesn't.
Would you want to switch the dependency to a gtihub tag?

Reply:
> It works, but it's quite verbose.

We could do a patch instead. Doesn't really matter to be fully honest.

> Github has tests folder, and its 0.7 release has tests, but Pypi package doesn't.
> Would you want to switch the dependency to a gtihub tag?

Yes please.